src/pages/_sections/Ticket.tsx

```diff
@@ -20,11 +21,20 @@ const sponsors = [
   }
 ];
 
-const downloadTicket = async (elementRef: RefObject<HTMLElement>) => {
+const downloadTicket = async (elementRef: RefObject<HTMLElement>, ticketId: string | null, providerId: string | null) => {
   // TODO: Send Generated Image to Supabase
-  if (elementRef.current) {
+  if (elementRef.current && ticketId && providerId) {
     try {
+      const img = await toBlob(elementRef.current);
       const dataUrl = await toPng(elementRef.current);
+
+      if (!img) {
+        console.error(); // TODO: Alert
+        return;
+      }
+
+      await uploadTicket(providerId, ticketId, img);
```

Comment:
@afordigital @Zyruks @jarrisondev do you think it is necessary to persist the ticket if it only downloads the image (taking into account that downloading does not require persisting)?

Reply:
You mean you don't need to save it because it will always make the call to the database?

Reply:
I mean, there are 2 buttons "Compartir" and "Descargar Ticket", the functionality to persist the image for later use as OG only serves if you want to share it on a social network (in this case in the "Compartir" button via Twitter), the action of "Descargar Ticket" is not necessary the functionality to persist and OG.

In my case, I do not have Twitter, but if I want to download the ticket (to download, it is only to make a "screenshot" of the ticket and save it in the pc), in these cases it is not necessary to save in DB or anything.